rasa/utils/tensorflow/models.py

```diff
@@ -585,88 +486,11 @@ def batch_to_model_data_format(
 
         return batch_data
 
-    @staticmethod
-    def linearly_increasing_batch_size(
-        epoch: int, batch_size: Union[List[int], int], epochs: int
-    ) -> int:
-        """Linearly increase batch size with every epoch.
-
-        The idea comes from https://arxiv.org/abs/1711.00489.
-        """
-
-        if not isinstance(batch_size, list):
-            return int(batch_size)
-
-        if epochs > 1:
-            return int(
-                batch_size[0] + epoch * (batch_size[1] - batch_size[0]) / (epochs - 1)
-            )
-        else:
-            return int(batch_size[0])
-
-    def _write_model_summary(self):
-        total_number_of_variables = np.sum(
-            [np.prod(v.shape) for v in self.trainable_variables]
-        )
-        layers = [
-            f"{layer.name} ({layer.dtype.name}) "
-            f"[{'x'.join(str(s) for s in layer.shape)}]"
-            for layer in self.trainable_variables
-        ]
-        layers.reverse()
-
-        with open(self.model_summary_file, "w") as file:
-            file.write("Variables: name (type) [shape]\n\n")
-            for layer in layers:
-                file.write(layer)
-                file.write("\n")
-            file.write("\n")
-            file.write(f"Total size of variables: {total_number_of_variables}")
-
-    def compile(self, *args, **kwargs) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
-        )
-
-    def evaluate(self, *args, **kwargs) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
-        )
-
-    def test_on_batch(self, *args, **kwargs) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
-        )
-
-    def predict_on_batch(self, *args, **kwargs) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
-        )
-
-    def fit_generator(self, *args, **kwargs) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
-        )
-
-    def evaluate_generator(self, *args, **kwargs) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
-        )
-
-    def predict_generator(self, *args, **kwargs) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
-        )
-
-    def call(self, *args, **kwargs) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
-        )
+    def call(self, inputs, training=None, mask=None):
+        pass
```

Comment:
what is this method?

Reply:
training fails if we don't include it, wanted to check why, but forgot to do it. will do and add a comment

Reply:
> raise NotImplementedError('When subclassing the `Model` class, you should '
                              'implement a `call` method.')

The method seems to be called during `build`. We don't need it, but we nevertheless need to overwrite it.

Reply:
it seems it is the main way to subclass keras model, I guess it is called in `train_step` and `predict_step`, but since you overwrote them, it is not used effectively. however, will the model be still called in build?